### PR TITLE
Feature add map filter

### PR DESCRIPTION
**Pull request by mrockingham**
# Description

Added Map to view incidents. Added search filter to search through available incidents. Added incident view to  view all incidents, clusters of incidents, or filtered incidents.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [ ] Complete, tested, ready to review and merge
- [x] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [ ] Test A
- [ ] Test B

# Checklist

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

### CODE REVIEW

File: src/components/Main.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react'
+import MapContainer from '../components/map/MapContainer'
+import { Layout } from 'antd';
+import Store from '../components/Store'
```
Comment (jlong5795) [inline src/components/Main.js:4]: Let's talk about removing this. There is a solution that shouldn't need an additional dependency or component files.

File: src/components/Store.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import React, { createContext, useState } from 'react';
```
Comment (jlong5795) [inline src/components/Store.js:1]: Same as comment on Main.js

File: src/components/map/MapSearch.js [hunk 3]
```diff
@@ -21,16 +31,26 @@ function callback(key) {
...
+    newViewport => setViewport(newViewport),
+    []
+  );
   const [filterDataList, setFilterDataList] = useState([]);
   const [searchText, setSearchText] = useState('');
-  const [long, setLong] = useState();
-  const [lat, setLat] = useState();
-  const [viewport, setViewport] = useState();
-  const [activeFilters, setActiveFilters] = useState ();
+  const [long, setLong] = useContext(ContextLong);;
```
Comment (jlong5795) [inline src/components/map/MapSearch.js:42]: I still think we should be using a state wrapper that isn't dependent upon an additional dependency but once we do that we should look at these states and see which of them change together and therefore should just be on one object together.

**Review by jlong5795 — CHANGES_REQUESTED** [review]
Let's meet and talk about this context vs wrapper situation.

**Review by jstewart8053 — APPROVED** [review]

**Comment by jlong5795** [comment]
Please update PR body description to describe what this PR accomplishes.

File: src/components/map/MapSearch.js [hunk 4]
```diff
@@ -107,37 +127,114 @@ const MapSearch = () => {
...
+              >
+                <h2 className='incident-header'>{lastIncident?.title}</h2>
+                <h3 className='incident-location'> {lastIncident?.city}, {lastIncident?.state}</h3>
+                
+                <h3 className='incident-discription'>{lastIncident?.desc}</h3>
+                
+              </div>
+              ): (
+
+                !incidentsOfInterest ? filterDataList.map((data, index) => {
```
Comment (jlong5795) [inline src/components/map/MapSearch.js:189]: I feel like we need to simplify this component. Perhaps parts can be extracted to separate components. Let's discuss.

**Review by jlong5795 — APPROVED** [review]

**Pull request merged**